pkg/execution/state/redis_state/lua/queue/partitionPeek.lua

```diff
@@ -22,21 +22,12 @@ if count > limit and sequential == 0 then
 	offset = math.random((count-limit)+1) - 1
 end
 
-local items = redis.call("ZRANGE", partitionIndex, "-inf", peekUntil, "BYSCORE", "LIMIT", offset, limit)
-if #items == 0 then
+local partitionIds = redis.call("ZRANGE", partitionIndex, "-inf", peekUntil, "BYSCORE", "LIMIT", offset, limit)
+if #partitionIds == 0 then
 	return {}
 end
 
-local potentiallyMissingPartitions = redis.call("HMGET", partitionKey, unpack(items))
-local missingPartitionIds = {}
-local validPartitions = {}
+local potentiallyMissingPartitions = redis.call("HMGET", partitionKey, unpack(partitionIds))
 
-for i, partition in ipairs(potentiallyMissingPartitions) do
```

Comment:
We moved from filtering out missing partitions in Lua to doing it in Go itself. This should speed up Lua scripts considerably.